### PR TITLE
Fix set system setting failure once it cached

### LINKED ISSUE #22332: Toggling  Picture and Avatar Configuration switches have no effect
### Description

Toggling the Avatar switches in admin/config URL does not change the configuration as requested: reloading the page moves the switches back to their original position.

I verified the request is sent to the server, the request is also logged by the server:
> 2023/01/04 00:04:41 [63b4b489] router: completed POST /admin/config for 192.168.1.82:44014, 200 OK in 2.6ms @ admin/config.go:193(admin.ChangeConfig)

BUT there's no change in the database (PostgreSQL) and reloading the page shows it (switches are back to their original position)

### Gitea Version

1.19.0+dev-264-g32e863439 P

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

from build tree. Cannot reproduce on try.gitea.io only because I'm not admin there, but if you grant me admin (strk) I can try

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by lunny**
Unfortunately, #22295 introduced a bug that when set a cached system setting, it will not affect.
This PR make sure to remove the cache key when updating a system setting.

Fix #22332

### CODE REVIEW

File: models/system/setting.go [hunk 1]
```diff
@@ -184,15 +184,8 @@ func SetSettingNoVersion(key, value string) error {
...
-		return setting.SettingValue, upsertSettingValue(strings.ToLower(setting.SettingKey), setting.SettingValue, setting.Version)
-	})
-	if err != nil {
-		return err
-	}
-
-	setting.Version++
-	return nil
+	cache.Remove(genSettingCacheKey(setting.SettingKey))
+	return upsertSettingValue(strings.ToLower(setting.SettingKey), setting.SettingValue, setting.Version)
```
Comment (delvh) [inline models/system/setting.go]: Why not stay at `cache.Set`?
What's the benefit of forcefully removing it from the cache?

File: models/system/setting_test.go [hunk 2]
```diff
@@ -37,6 +37,11 @@ func TestSettings(t *testing.T) {
 	err = system.SetSetting(updatedSetting)
 	assert.NoError(t, err)
 
+	// make sure it's not cached
+	updatedSetting = &system.Setting{SettingKey: keyName, SettingValue: "100", Version: newSetting.Version}
+	err = system.SetSetting(updatedSetting)
+	assert.NoError(t, err)
```
Comment (delvh) [inline models/system/setting_test.go]: Isn't there an `assert` missing?
At which place do you test that it is not cached?

File: models/system/setting.go (hunk 1 above)
Comment (lunny) [reply models/system/setting.go]: Resolved. Now cache will be updated after database operation succeed.

File: models/system/setting_test.go (hunk 2 above)
Comment (lunny) [reply models/system/setting_test.go]: This lines removed. Please review again.

Comment (lunny) [reply models/system/setting_test.go]: This lines removed. Please review again.

File: models/system/setting.go [hunk 3]
```diff
@@ -88,7 +88,7 @@ func GetSettingNoCache(key string) (*Setting, error) {
 	if len(v) == 0 {
 		return nil, ErrSettingIsNotExist{key}
 	}
-	return v[key], nil
+	return v[strings.ToLower(key)], nil
```
Comment (delvh) [inline models/system/setting.go:91]: I have the suspicion that we lowercase to often at the moment.

**Review by delvh — APPROVED** [review]

File: models/system/setting.go (hunk 3 above)
Comment (lunny) [reply models/system/setting.go:91]: Yes, but are there better idea to reduce them?

**Review by wolfogre — APPROVED** [review]

File: models/system/setting.go (hunk 3 above)
Comment (delvh) [reply models/system/setting.go:91]: Extracting internal methods that assume the key is lowercase already?
However, I approved this PR as this is off-topic and simply a performance improvement.

**Pull request merged**